frontend_lib/src/component/ConfirmPopup/ConfirmPopup.jsx

```diff
@@ -0,0 +1,50 @@
+// require('./ConfirmPopup.styl') // see https://github.com/tracim/tracim/issues/1156
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import CardPopup from '../CardPopup/CardPopup.jsx'
+import { translate } from 'react-i18next'
+
+const ConfirmPopup = (props) => (
+  <CardPopup
+    customClass='confirm_popup'
+    customHeaderClass='primaryColorBg'
+    onClose={props.onCancel}
+  >
+    <div className='confirm_popup__body'>
+      <div className='confirm_popup__body__msg'>{props.msg || props.t('Are you sure?')}</div>
+      <div className='confirm_popup__body__btn'>
+        <button
+          type='button'
+          className='btn outlineTextBtn primaryColorBorder primaryColorFont nohover'
+          onClick={props.onCancel}
+        >
+          {props.t('Cancel')}
+        </button>
+        <button
+          type='button'
+          className='btn highlightBtn primaryColorBg primaryColorDarkenBgHover'
+          onClick={props.onConfirm}
+        >
+          {props.confirmLabel || props.t('Confirm')}
+        </button>
+      </div>
+    </div>
+  </CardPopup>
+)
+
+ConfirmPopup.propType = {
+  confirmLabel: PropTypes.string,
+  onConfirm: PropTypes.func,
+  onCancel: PropTypes.func,
+  msg: PropTypes.string
+}
+
+ConfirmPopup.defaultProps = {
+  confirmLabel: '',
+  onConfirm: null,
+  onCancel: null,
```

Comment:
the propTypes expects a function but you give `null` as default value

Reply:
it expects an optional function:
>   You can declare that a prop is a specific JS type. By default, these are all optional.

https://reactjs.org/docs/typechecking-with-proptypes.html

The code handles null correctly.

Reply:
if you ask for a function that means you will bind some handlers to these parameters
If it is null, it might be converted to `null()` which will trigger a js error.
Also, optional parameters will be `undefined` if not given without a default value

Reply:
But since it is optional, I check before use.

Reply:
mmh I don't see where

Reply:
onClick={`props.onCancel`} will be `onClick={null}` if `props.onCancel` is `null`, which is fine.

Anyway, Il make these props required, it does not make sense to not set them.

Reply:
done.